pkg/workflows/wasm/workflowLib/transmitter.go

```diff
@@ -0,0 +1,74 @@
+package workflowLib
+
+import (
+	"context"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/workflows/wasm/workflowLib/pb"
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/ocr3types"
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+)
+
+var _ ocr3types.ContractTransmitter[struct{}] = (*Transmitter)(nil)
+
+// Transmitter is a custom transmitter for the OCR3 capability.
+// When called it will transmit DonTime requests back to the caller
+// and handle deletion of finished executionIDs.
+type Transmitter struct {
+	lggr      logger.Logger
+	store     *DonTimeStore
+	batchSize int
+}
+
+func NewTransmitter(lggr logger.Logger, store *DonTimeStore, batchSize int) *Transmitter {
+	return &Transmitter{lggr: lggr, store: store, batchSize: batchSize}
+}
+
+func (t *Transmitter) Transmit(ctx context.Context, _ types.ConfigDigest, _ uint64, r ocr3types.ReportWithInfo[struct{}], _ []types.AttributedOnchainSignature) error {
+	if err := ctx.Err(); err != nil {
+		return err
+	}
+
+	outcome := &pb.Outcome{}
+	if err := proto.Unmarshal(r.Report, outcome); err != nil {
+		return err
+	}
+
+	for id, observedDonTimes := range outcome.ObservedDonTimes {
+		t.store.SetDonTimes(id, observedDonTimes.Timestamps)
+	}
+	t.store.SetLastObservedDonTime(outcome.Timestamp)
+
+	requests, err := t.store.Requests.FirstN(t.batchSize)
```

Comment:
This is not right, you should iterate over what you have in your outcome, not over FirstN again.